### DISCUSSION TITLE
Is it possible to set custom directory for temp files for StreamWriter instead of default directory?

### DISCUSSION THREAD

**Discussion by stasvo** [qax-os/excelize#1449] [Q&A]
Is it possible to set custom directory for temporary files instead of use default directory of OS when work with StreamWriter?

**Comment by xuri** [comment]
Thanks for your suggestions. I will consider adding support in the future for this.

**Reply by Denvormine** [reply]
Hi, how hard it could be to add this feature? We're in need of it because we might be generating a lot of big excels and tmp directory can run out of space. It's possible to consider making pull request with this feature but I want to know your thoughts on it's implementation.

**Comment by xuri** [comment]
Currently, you can use the `TMPDIR=/custom_directory` environment or set the `os.Setenv("TMPDIR", "/custom_directory")` change the default temp directory.
